lib/player.js

```diff
@@ -1598,6 +1598,52 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     });
   }
 
+  /**
+   * Unloads the currently playing stream, if any, and returns a PreloadManager
+   * that contains the loaded manifest of that asset, if any.
+   * Allows for the asset to be re-loaded by this player faster, in the future.
+   * When in src= mode, this unloads but does not make a PreloadManager.
+   *
+   * @param {boolean=} initializeMediaSource
+   * @param {boolean=} keepAdManager
+   * @return {!Promise.<?shaka.media.PreloadManager>}
+   * @export
+   */
+  async unloadAndSavePreload(
+      initializeMediaSource = true, keepAdManager = false) {
+    let preloadManager = null;
+    if (this.manifest_ && this.parser_ && this.parserFactory_ &&
+        this.assetUri_) {
+      let startTime = this.video_.currentTime;
+      if (this.isLive()) {
+        startTime = null;
+      }
+      // We have enough information to make a PreloadManager!
+      const startTimeOfLoad = Date.now() / 1000;
+      preloadManager = await this.makePreloadManager_(
+          this.assetUri_,
+          startTime,
+          this.mimeType_,
+          startTimeOfLoad,
+          /* allowPrefetch= */ true,
+          /* disableVideo= */ false,
+          /* allowMakeAbrManager= */ false);
+      preloadManager.attachManifest(
+          this.manifest_, this.parser_, this.parserFactory_);
+      preloadManager.attachAbrManager(this.abrManager_);
+      preloadManager.attachAdaptationSetCriteria(
+          this.currentAdaptationSetCriteria_);
+      preloadManager.start();
+      // Null the manifest and manifestParser, so that they won't be shut down
+      // during unload and will continue to live inside the preloadManager.
+      this.manifest_ = null;
+      this.parser_ = null;
+      this.parserFactory_ = null;
```

Comment:
You will also want to set `this.abrManager_` to null, so that the player doesn't call `this.abrManager_.stop()` while unloading.